### PR TITLE
Refactor Board Class

### PR DESCRIPTION
**Pull request by jbranham1**

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -29,30 +29,32 @@ def valid_coordinate?(coordinate)
...
+  end
+
+  def letters(coordinates)
     (coordinates[0][0]..coordinates[-1][0]).to_a
   end
 
-  def set_numbers(coordinates)
+  def numbers(coordinates)
     (coordinates[0][1]..coordinates[-1][1]).to_a
   end
```
Comment (amsmyth1) [inline lib/board.rb:51]: The letters and numbers methods are going to work so well if we try iteration 4. These lines of code really make me happy.

File: lib/board.rb [hunk 2]
```diff
@@ -29,30 +29,32 @@ def valid_coordinate?(coordinate)
...
-    numbers = set_numbers(coordinates)
-    if letters.all?(coordinates[0][0])
-      numbers.size == coordinates.size
-    elsif numbers.all?(coordinates[0][1])
-      letters.size == coordinates.size
+    if letters(coordinates).all?(coordinates[0][0])
+      numbers(coordinates).size == coordinates.size
+    elsif numbers(coordinates).all?(coordinates[0][1])
+      letters(coordinates).size == coordinates.size
     else
```
Comment (amsmyth1) [inline lib/board.rb:58]: This is a lot easier to read, nicely done.

File: lib/board.rb (hunk 1 above)
Comment (jbranham1) [reply lib/board.rb:51]: Yay! Glad you like it!

**Review by amsmyth1 — APPROVED** [review]
looks good, I like the away you refactored the letters/numbers methods to make the validation clearer.

**Pull request merged**